app/lib/domains.server.ts

```diff
@@ -0,0 +1,148 @@
+import * as dns from './dns.server';
+import { RecordStatus } from '@prisma/client';
+import { prisma } from '~/db.server';
+import type { RecordType } from './dns.server';
+
+interface DNSProp {
+  type: RecordType;
+  name: string;
+  value: string;
+}
+
+interface DBProp {
```

Comment:
You can use the `Record` type created by Prisma.  I exposes a TypeScript definition.

```ts
import type { Record } from '@prisma/client';
```

Reply:
`Record type` definition forces me to add all the properties to the data. 
```js
    const data: Record = {
      type: RecordType.A,
      name: domain,
      value,
      id: 1,
      username,
      description,
      course,
      ports,
      createdAt: new Date(),
      updatedAt: new Date(),
      expiresAt: changeMonthFromNow(6),
      status: RecordStatus.pending,
    };
```
Is it the way it should be or is there a way that I can make properties optional?

Reply:
You should be able to do optional args, if the types are actually optional.  Consider:

```ts
type A = {
   one: string;
   two?: string;
}

type B = {
   three: string;
   four?: string;
}

type C = A & B;

let c: C = {
   one: 'one',
   three: 'three'
};
```

Reply:
Oh, you mean the Prisma type:

```
model Record {
  id          Int          @id @default(autoincrement())
  username    String
  name        String
  type        RecordType
  value       String
  description String?
  course      String?
  ports       String?
  createdAt   DateTime     @default(now())
  updatedAt   DateTime     @updatedAt
  expiresAt   DateTime
  status      RecordStatus
  user        User         @relation(fields: [username], references: [username])
}
```

Here you can see that only a few are optional.

Reply:
The other things you can do that might help:

- Create a type that has all the properties as optional: https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype
- Pick the specific properties you want, but not the others: https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys
- Omit the specific properties you want, but not the others: https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys

Using these, you should be able to use a mix of intersect, pick, omit to produce what you need.